src/moments.jl

```diff
@@ -58,195 +68,231 @@ function var!(R::AbstractArray, A::RealArray, wv::WeightVec, dim::Int; mean=noth
+    mean_and_var(x, [wv::AbstractWeights], [dim]) -> (mean, var)
 
 Return the mean and variance of a real-valued array `x`, optionally over a dimension
 `dim`, as a tuple. A weighting vector `wv` can be specified to weight the estimates.
 The weights are assumed to be frequency weights, also called case weights.
 """
-mean_and_var(A::RealArray) = (m = mean(A); (m, varm(A, m)))
+function mean_and_var(A::RealArray; corrected=false)
+    m = mean(A)
+    v = varm(A, m; corrected=corrected)
+    m, v
+end
 
 """
-    mean_and_std(x, [wv::WeightVec], [dim]) -> (mean, std)
+    mean_and_std(x, [wv::AbstractWeights], [dim]) -> (mean, std)
 
 Return the mean and standard deviation of a real-valued array `x`, optionally
 over a dimension `dim`, as a tuple. A weighting vector `wv` can be specified
 to weight the estimates. The weights are assumed to be frequency weights, also
 called case weights.
 """
-mean_and_std(A::RealArray) = (m = mean(A); (m, stdm(A, m)))
+function mean_and_std(A::RealArray; corrected=false)
+    m = mean(A)
+    s = stdm(A, m; corrected=corrected)
+    m, s
+end
 
-mean_and_var(A::RealArray, wv::WeightVec) = (m = mean(A, wv); (m, varm(A, wv, m)))
-mean_and_std(A::RealArray, wv::WeightVec) = (m = mean(A, wv); (m, stdm(A, wv, m)))
+function mean_and_var(A::RealArray, wv::AbstractWeights; corrected=false)
+    m = mean(A, wv)
+    v = varm(A, wv, m; corrected=corrected)
+    m, v
+end
 
-mean_and_var(A::RealArray, dim::Int) = (m = mean(A, dim); (m, varm(A, m, dim)))
-mean_and_std(A::RealArray, dim::Int) = (m = mean(A, dim); (m, stdm(A, m, dim)))
+function mean_and_std(A::RealArray, wv::AbstractWeights; corrected=false)
+    m = mean(A, wv)
+    s = stdm(A, wv, m; corrected=corrected)
+    m, s
+end
 
-mean_and_var(A::RealArray, wv::WeightVec, dim::Int) = (m = mean(A, wv, dim); (m, varm(A, wv, m, dim)))
-mean_and_std(A::RealArray, wv::WeightVec, dim::Int) = (m = mean(A, wv, dim); (m, stdm(A, wv, m, dim)))
+function mean_and_var(A::RealArray, dim::Int; corrected=false)
+    m = mean(A, dim)
+    v = varm(A, m, dim; corrected=corrected)
+    m, v
+end
 
+function mean_and_std(A::RealArray, dim::Int; corrected=false)
+    m = mean(A, dim)
+    s = stdm(A, m, dim; corrected=corrected)
+    m, s
+end
+
+function mean_and_var(A::RealArray, wv::AbstractWeights, dim::Int; corrected=false)
+    m = mean(A, wv, dim)
+    v = varm(A, wv, m, dim; corrected=corrected)
+    m, v
+end
+
+function mean_and_std(A::RealArray, wv::AbstractWeights, dim::Int; corrected=false)
+    m = mean(A, wv, dim)
+    s = stdm(A, wv, m, dim; corrected=corrected)
+    m, s
+end
 
 ##### General central moment
+"""
+    moment(v, k, [wv::AbstractWeights], m=mean(v); corrected=false)
 
-function _moment2(v::RealArray, m::Real)
+Return the `k`th order central moment of a real-valued array `v`, optionally
+specifying a weighting vector `wv` and a center `m`.
+"""
+moment(v::RealArray, k::Int, args...; kwargs...) = moment(v, Val{k}, args...; kwargs...)
```

Comment:
That `Val` change is completely unrelated to this PR, please leave it out. I also doubt it will increase performance (rather the contrary).